ICE: `cannot convert ReLateParam to a region vid`

<!--
[31mICE[0m: Rustc ./a.rs '' 'error: internal compiler error: compiler/rustc_borrowck/src/universal_regions.rs:888:36: cannot convert `ReLateParam(DefId(0:16 ~ a[ff78]::get_invariant_ref), BrNamed(DefId(0:17 ~ a[ff78]::get_invariant_ref::'a), 'a))` to a region vid', 'error: internal compiler error: compiler/rustc_borrowck/src/universal_regions.rs:888:36: cannot convert `ReLateParam(DefId(0:16 ~ a[ff78]::get_invariant_ref), BrNamed(DefId(0:17 ~ a[ff78]::get_invariant_ref::'a), 'a))` to a region vid'
File: /tmp/im/a.rs
-->

auto-reduced (treereduce-rust):
````rust
pub struct InvariantRef<'a, T: ?Sized>(&'a T);

impl<'a, T: ?Sized> InvariantRef<'a, T> {
    pub const fn new(r: &'a T) -> Self {}
}

fn get_invariant_ref<'a>() -> InvariantRef<'static, _> {
    const { InvariantRef::<'a, ()>::new(&()) }
}

````

original:
````rust
//@ run-pass

use std::marker::PhantomData;

// rust-lang/rust#78174: ICE: "cannot convert ReErased to a region vid"
fn issue_78174() {
    let foo = const { "foo" };
    assert_eq!(foo, "foo");
}

pub struct InvariantRef<'a, T: ?Sized>(&'a T, PhantomData<&'a mut &'a T>);

impl<'a, T: ?Sized> InvariantRef<'a, T> {
    pub const fn new(r: &'a T) -> Self {
        InvariantRef(r, PhantomData)
    }
}

fn get_invariant_ref<'a>() -> InvariantRef<'static, _> {
    const { InvariantRef::<'a, ()>::new(&()) }
}

fn get_invariant_ref2<'a>() -> InvariantRef<'a, ()> {
    // Try some type inference
    const { InvariantRef::new(&()) }
}

fn main() {
    issue_78174();
    get_invariant_ref();
    get_invariant_ref2();
}

````

Version information
````
rustc 1.86.0-nightly (cd805f09f 2025-01-21)
binary: rustc
commit-hash: cd805f09ffbfa3896c8f50a619de9b67e1d9f3c3
commit-date: 2025-01-21
host: x86_64-unknown-linux-gnu
release: 1.86.0-nightly
LLVM version: 19.1.7

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/cd805f09ffbfa3896c8f50a619de9b67e1d9f3c3/compiler/rustc_borrowck/src/universal_regions.rs#L882-L894

Command:
`/home/matthias/.rustup/toolchains/master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0601]: `main` function not found in crate `mvce`
 --> /tmp/icemaker_global_tempdir.sXnYG42JCCJ4/rustc_testrunner_tmpdir_reporting.qVKGb9NYPlaR/mvce.rs:9:2
  |
9 | }
  |  ^ consider adding a `main` function to `/tmp/icemaker_global_tempdir.sXnYG42JCCJ4/rustc_testrunner_tmpdir_reporting.qVKGb9NYPlaR/mvce.rs`

error[E0121]: the placeholder `_` is not allowed within types on item signatures for return types
 --> /tmp/icemaker_global_tempdir.sXnYG42JCCJ4/rustc_testrunner_tmpdir_reporting.qVKGb9NYPlaR/mvce.rs:7:53
  |
7 | fn get_invariant_ref<'a>() -> InvariantRef<'static, _> {
  |                               ----------------------^-
  |                               |                     |
  |                               |                     not allowed in type signatures
  |                               help: replace with the correct return type: `InvariantRef<'_, ()>`

error[E0308]: mismatched types
 --> /tmp/icemaker_global_tempdir.sXnYG42JCCJ4/rustc_testrunner_tmpdir_reporting.qVKGb9NYPlaR/mvce.rs:4:35
  |
4 |     pub const fn new(r: &'a T) -> Self {}
  |                  ---              ^^^^ expected `InvariantRef<'_, T>`, found `()`
  |                  |
  |                  implicitly returns `()` as its body has no tail or `return` expression
  |
  = note: expected struct `InvariantRef<'a, T>`
          found unit type `()`

error: internal compiler error: compiler/rustc_borrowck/src/universal_regions.rs:888:36: cannot convert `ReLateParam(DefId(0:12 ~ mvce[18ee]::get_invariant_ref), BrNamed(DefId(0:13 ~ mvce[18ee]::get_invariant_ref::'a), 'a))` to a region vid


thread 'rustc' panicked at compiler/rustc_borrowck/src/universal_regions.rs:888:36:
Box<dyn Any>
stack backtrace:
   0:     0x703ddf2fabfa - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::he8f87afa3ba953c7
   1:     0x703ddfa12de6 - core::fmt::write::hd1a70dafa054ef6e
   2:     0x703de0932191 - std::io::Write::write_fmt::hb7487c14cde15a5b
   3:     0x703ddf2faa52 - std::sys::backtrace::BacktraceLock::print::hccabcb6160db5eae
   4:     0x703ddf2fced2 - std::panicking::default_hook::{{closure}}::h8c6502dfd308289c
   5:     0x703ddf2fcd5a - std::panicking::default_hook::hb4c1963f11244643
   6:     0x703dde459a8b - std[59f0843774866faa]::panicking::update_hook::<alloc[ba5425907d200e29]::boxed::Box<rustc_driver_impl[d01b6faf608d79fd]::install_ice_hook::{closure#1}>>::{closure#0}
   7:     0x703ddf2fda13 - std::panicking::rust_panic_with_hook::hc3d8183c3b9ea233
   8:     0x703dde4948a1 - std[59f0843774866faa]::panicking::begin_panic::<rustc_errors[33b093a44d219c0d]::ExplicitBug>::{closure#0}
   9:     0x703dde4897a6 - std[59f0843774866faa]::sys::backtrace::__rust_end_short_backtrace::<std[59f0843774866faa]::panicking::begin_panic<rustc_errors[33b093a44d219c0d]::ExplicitBug>::{closure#0}, !>
  10:     0x703dde4862b9 - std[59f0843774866faa]::panicking::begin_panic::<rustc_errors[33b093a44d219c0d]::ExplicitBug>
  11:     0x703dde49e7d1 - <rustc_errors[33b093a44d219c0d]::diagnostic::BugAbort as rustc_errors[33b093a44d219c0d]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
  12:     0x703ddea800e3 - rustc_middle[d9c304bb3c99a636]::util::bug::opt_span_bug_fmt::<rustc_span[79f25feaa0a05865]::span_encoding::Span>::{closure#0}
  13:     0x703ddea658da - rustc_middle[d9c304bb3c99a636]::ty::context::tls::with_opt::<rustc_middle[d9c304bb3c99a636]::util::bug::opt_span_bug_fmt<rustc_span[79f25feaa0a05865]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  14:     0x703ddea6576b - rustc_middle[d9c304bb3c99a636]::ty::context::tls::with_context_opt::<rustc_middle[d9c304bb3c99a636]::ty::context::tls::with_opt<rustc_middle[d9c304bb3c99a636]::util::bug::opt_span_bug_fmt<rustc_span[79f25feaa0a05865]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  15:     0x703ddcbd4270 - rustc_middle[d9c304bb3c99a636]::util::bug::bug_fmt
  16:     0x703de03b2479 - <rustc_borrowck[418e11b174dea340]::type_check::constraint_conversion::ConstraintConversion>::convert
  17:     0x703de03b158e - <rustc_borrowck[418e11b174dea340]::type_check::TypeChecker>::prove_closure_bounds
  18:     0x703ddfc6e6ca - <rustc_borrowck[418e11b174dea340]::type_check::TypeChecker>::typeck_mir
  19:     0x703de0ab7e93 - rustc_borrowck[418e11b174dea340]::type_check::type_check
  20:     0x703ddfb0777e - rustc_borrowck[418e11b174dea340]::nll::compute_regions
  21:     0x703de0adbb87 - rustc_borrowck[418e11b174dea340]::do_mir_borrowck
  22:     0x703de0acebaf - rustc_query_impl[51e2c968ef770bf4]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[51e2c968ef770bf4]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[d9c304bb3c99a636]::query::erase::Erased<[u8; 8usize]>>
  23:     0x703ddfddbc0e - rustc_query_system[62ef735fec9db900]::query::plumbing::try_execute_query::<rustc_query_impl[51e2c968ef770bf4]::DynamicConfig<rustc_data_structures[ab2428ff1810be63]::vec_cache::VecCache<rustc_span[79f25feaa0a05865]::def_id::LocalDefId, rustc_middle[d9c304bb3c99a636]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[62ef735fec9db900]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[51e2c968ef770bf4]::plumbing::QueryCtxt, false>
  24:     0x703ddfddb70d - rustc_query_impl[51e2c968ef770bf4]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  25:     0x703ddfdd4900 - rustc_interface[511fe094a4999c01]::passes::run_required_analyses
  26:     0x703de092d99e - rustc_interface[511fe094a4999c01]::passes::analysis
  27:     0x703de092d96f - rustc_query_impl[51e2c968ef770bf4]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[51e2c968ef770bf4]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[d9c304bb3c99a636]::query::erase::Erased<[u8; 0usize]>>
  28:     0x703de0a083d5 - rustc_query_system[62ef735fec9db900]::query::plumbing::try_execute_query::<rustc_query_impl[51e2c968ef770bf4]::DynamicConfig<rustc_query_system[62ef735fec9db900]::query::caches::SingleCache<rustc_middle[d9c304bb3c99a636]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[51e2c968ef770bf4]::plumbing::QueryCtxt, false>
  29:     0x703de0a0810e - rustc_query_impl[51e2c968ef770bf4]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  30:     0x703de09a1169 - rustc_interface[511fe094a4999c01]::passes::create_and_enter_global_ctxt::<core[ee856ee324d81d9a]::option::Option<rustc_interface[511fe094a4999c01]::queries::Linker>, rustc_driver_impl[d01b6faf608d79fd]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}
  31:     0x703de09ef950 - rustc_interface[511fe094a4999c01]::interface::run_compiler::<(), rustc_driver_impl[d01b6faf608d79fd]::run_compiler::{closure#0}>::{closure#1}
  32:     0x703de0940076 - std[59f0843774866faa]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[511fe094a4999c01]::util::run_in_thread_with_globals<rustc_interface[511fe094a4999c01]::util::run_in_thread_pool_with_globals<rustc_interface[511fe094a4999c01]::interface::run_compiler<(), rustc_driver_impl[d01b6faf608d79fd]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  33:     0x703de093fd59 - <<std[59f0843774866faa]::thread::Builder>::spawn_unchecked_<rustc_interface[511fe094a4999c01]::util::run_in_thread_with_globals<rustc_interface[511fe094a4999c01]::util::run_in_thread_pool_with_globals<rustc_interface[511fe094a4999c01]::interface::run_compiler<(), rustc_driver_impl[d01b6faf608d79fd]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[ee856ee324d81d9a]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  34:     0x703de093f4ef - std::sys::pal::unix::thread::Thread::new::thread_start::h66685cea0b34fc94
  35:     0x703ddaca339d - <unknown>
  36:     0x703ddad2849c - <unknown>
  37:                0x0 - <unknown>

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.86.0-nightly (cd805f09f 2025-01-21) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [mir_borrowck] borrow-checking `get_invariant_ref`
#1 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 4 previous errors

Some errors have detailed explanations: E0121, E0308, E0601.
For more information about an error, try `rustc --explain E0121`.

```

</p>
</details>

<!--
query stack:
#0 [mir_borrowck] borrow-checking `get_invariant_ref`
#1 [analysis] running analysis passes on this crate

-->

--- a/compiler/rustc_hir_analysis/src/collect.rs
+++ b/compiler/rustc_hir_analysis/src/collect.rs
@@ -1063,6 +1063,69 @@ fn lower_fn_sig_recovering_infer_ret_ty<'tcx>(
     )
 }
 
+/// Convert `ReLateParam`s in `value` back into `ReBound`s so the returned
+/// `PolyFnSig` properly binds all late-bound vars.
+fn late_param_regions_to_bound<'tcx, T>(
+    tcx: TyCtxt<'tcx>,
+    scope: DefId,
+    bound_vars: &'tcx ty::List<ty::BoundVariableKind<'tcx>>,
+    value: T,
+) -> T
+where
+    T: ty::TypeFoldable<TyCtxt<'tcx>>,
+{
+    fold_regions(tcx, value, |r, debruijn| match r.kind() {
+        ty::ReLateParam(lp) => {
+            // Should be in scope, otherwise inconsistency happens somewhere.
+            assert_eq!(lp.scope, scope);
+
+            let br = match lp.kind {
+                // These variants preserve the bound var index.
+                kind @ (ty::LateParamRegionKind::Anon(idx)
+                | ty::LateParamRegionKind::NamedAnon(idx, _)) => {
+                    let idx = idx as usize;
+                    let var = ty::BoundVar::from_usize(idx);
+
+                    let Some(ty::BoundVariableKind::Region(kind)) = bound_vars.get(idx).copied()
+                    else {
+                        bug!("unexpected late-bound region {kind:?} for bound vars {bound_vars:?}");
+                    };
+
+                    ty::BoundRegion { var, kind }
+                }
+
+                // For named regions, look up the corresponding bound var.
+                ty::LateParamRegionKind::Named(def_id) => bound_vars
+                    .iter()
+                    .enumerate()
+                    .find_map(|(idx, bv)| match bv {
+                        ty::BoundVariableKind::Region(kind @ ty::BoundRegionKind::Named(did))
+                            if did == def_id =>
+                        {
+                            Some(ty::BoundRegion { var: ty::BoundVar::from_usize(idx), kind })
+                        }
+                        _ => None,
+                    })
+                    .unwrap(),
+
+                ty::LateParamRegionKind::ClosureEnv => bound_vars
+                    .iter()
+                    .enumerate()
+                    .find_map(|(idx, bv)| match bv {
+                        ty::BoundVariableKind::Region(kind @ ty::BoundRegionKind::ClosureEnv) => {
+                            Some(ty::BoundRegion { var: ty::BoundVar::from_usize(idx), kind })
+                        }
+                        _ => None,
+                    })
+                    .unwrap(),
+            };
+
+            ty::Region::new_bound(tcx, debruijn, br)
+        }
+        _ => r,
+    })
+}
+
 fn recover_infer_ret_ty<'tcx>(
     icx: &ItemCtxt<'tcx>,
     infer_ret_ty: &'tcx hir::Ty<'tcx>,
@@ -1153,48 +1216,7 @@ fn recover_infer_ret_ty<'tcx>(
         fn_sig.abi,
     );
 
-    let fn_sig = fold_regions(tcx, fn_sig, |r, _| match r.kind() {
-        ty::ReLateParam(lp) if lp.scope == scope => {
-            let br = match lp.kind {
-                ty::LateParamRegionKind::Anon(idx) | ty::LateParamRegionKind::NamedAnon(idx, _) => {
-                    let idx = idx as usize;
-                    match bound_vars.get(idx).copied() {
-                        Some(ty::BoundVariableKind::Region(br_kind)) => Some(ty::BoundRegion {
-                            var: ty::BoundVar::from_usize(idx),
-                            kind: br_kind,
-                        }),
-                        _ => None,
-                    }
-                }
-                ty::LateParamRegionKind::Named(def_id) => {
-                    bound_vars.iter().enumerate().find_map(|(idx, bv)| match bv {
-                        ty::BoundVariableKind::Region(
-                            br_kind @ ty::BoundRegionKind::Named(did),
-                        ) if did == def_id => Some(ty::BoundRegion {
-                            var: ty::BoundVar::from_usize(idx),
-                            kind: br_kind,
-                        }),
-                        _ => None,
-                    })
-                }
-                ty::LateParamRegionKind::ClosureEnv => {
-                    bound_vars.iter().enumerate().find_map(|(idx, bv)| match bv {
-                        ty::BoundVariableKind::Region(
-                            br_kind @ ty::BoundRegionKind::ClosureEnv,
-                        ) => Some(ty::BoundRegion {
-                            var: ty::BoundVar::from_usize(idx),
-                            kind: br_kind,
-                        }),
-                        _ => None,
-                    })
-                }
-            };
-
-            br.map(|br| ty::Region::new_bound(tcx, ty::INNERMOST, br))
-                .unwrap_or_else(|| ty::Region::new_error(tcx, guar))
-        }
-        _ => r,
-    });
+    let fn_sig = late_param_regions_to_bound(tcx, scope, bound_vars, fn_sig);
 
     ty::Binder::bind_with_vars(fn_sig, bound_vars)
 }

--- a/compiler/rustc_hir_analysis/src/collect.rs
+++ b/compiler/rustc_hir_analysis/src/collect.rs
@@ -1063,18 +1063,17 @@ fn lower_fn_sig_recovering_infer_ret_ty<'tcx>(
     )
 }
 
-/// Convert `ReLateParam`s in `value` back into `ReBound`s so the returned
-/// `PolyFnSig` properly binds all late-bound vars.
+/// Convert `ReLateParam`s in `value` back into `ReBound`s and bind it with `bound_vars`.
 fn late_param_regions_to_bound<'tcx, T>(
     tcx: TyCtxt<'tcx>,
     scope: DefId,
     bound_vars: &'tcx ty::List<ty::BoundVariableKind<'tcx>>,
     value: T,
-) -> T
+) -> ty::Binder<'tcx, T>
 where
     T: ty::TypeFoldable<TyCtxt<'tcx>>,
 {
-    fold_regions(tcx, value, |r, debruijn| match r.kind() {
+    let value = fold_regions(tcx, value, |r, debruijn| match r.kind() {
         ty::ReLateParam(lp) => {
             // Should be in scope, otherwise inconsistency happens somewhere.
             assert_eq!(lp.scope, scope);
@@ -1123,7 +1122,9 @@ where
             ty::Region::new_bound(tcx, debruijn, br)
         }
         _ => r,
-    })
+    });
+
+    ty::Binder::bind_with_vars(value, bound_vars)
 }
 
 fn recover_infer_ret_ty<'tcx>(
@@ -1216,9 +1217,7 @@ fn recover_infer_ret_ty<'tcx>(
         fn_sig.abi,
     );
 
-    let fn_sig = late_param_regions_to_bound(tcx, scope, bound_vars, fn_sig);
-
-    ty::Binder::bind_with_vars(fn_sig, bound_vars)
+    late_param_regions_to_bound(tcx, scope, bound_vars, fn_sig)
 }
 
 pub fn suggest_impl_trait<'tcx>(

--- a/compiler/rustc_hir_analysis/src/collect.rs
+++ b/compiler/rustc_hir_analysis/src/collect.rs
@@ -1138,13 +1138,65 @@ fn recover_infer_ret_ty<'tcx>(
         );
     }
     let guar = diag.emit();
-    ty::Binder::dummy(tcx.mk_fn_sig(
+
+    // If we return a dummy binder here, we can ICE later in borrowck when it encounters
+    // `ReLateParam` regions (e.g. in a local type annotation) which weren't registered via the
+    // signature binder. See #135845.
+    let bound_vars = tcx.late_bound_vars(hir_id);
+    let scope = def_id.to_def_id();
+
+    let fn_sig = tcx.mk_fn_sig(
         fn_sig.inputs().iter().copied(),
         recovered_ret_ty.unwrap_or_else(|| Ty::new_error(tcx, guar)),
         fn_sig.c_variadic,
         fn_sig.safety,
         fn_sig.abi,
-    ))
+    );
+
+    let fn_sig = fold_regions(tcx, fn_sig, |r, _| match r.kind() {
+        ty::ReLateParam(lp) if lp.scope == scope => {
+            let br = match lp.kind {
+                ty::LateParamRegionKind::Anon(idx) | ty::LateParamRegionKind::NamedAnon(idx, _) => {
+                    let idx = idx as usize;
+                    match bound_vars.get(idx).copied() {
+                        Some(ty::BoundVariableKind::Region(br_kind)) => Some(ty::BoundRegion {
+                            var: ty::BoundVar::from_usize(idx),
+                            kind: br_kind,
+                        }),
+                        _ => None,
+                    }
+                }
+                ty::LateParamRegionKind::Named(def_id) => {
+                    bound_vars.iter().enumerate().find_map(|(idx, bv)| match bv {
+                        ty::BoundVariableKind::Region(
+                            br_kind @ ty::BoundRegionKind::Named(did),
+                        ) if did == def_id => Some(ty::BoundRegion {
+                            var: ty::BoundVar::from_usize(idx),
+                            kind: br_kind,
+                        }),
+                        _ => None,
+                    })
+                }
+                ty::LateParamRegionKind::ClosureEnv => {
+                    bound_vars.iter().enumerate().find_map(|(idx, bv)| match bv {
+                        ty::BoundVariableKind::Region(
+                            br_kind @ ty::BoundRegionKind::ClosureEnv,
+                        ) => Some(ty::BoundRegion {
+                            var: ty::BoundVar::from_usize(idx),
+                            kind: br_kind,
+                        }),
+                        _ => None,
+                    })
+                }
+            };
+
+            br.map(|br| ty::Region::new_bound(tcx, ty::INNERMOST, br))
+                .unwrap_or_else(|| ty::Region::new_error(tcx, guar))
+        }
+        _ => r,
+    });
+
+    ty::Binder::bind_with_vars(fn_sig, bound_vars)
 }
 
 pub fn suggest_impl_trait<'tcx>(

--- a/tests/crashes/135845.rs
+++ b/tests/crashes/135845.rs
@@ -1,6 +1,0 @@
-//@ known-bug: #135845
-struct S<'a, T: ?Sized>(&'a T);
-
-fn b<'a>() -> S<'static, _> {
-    S::<'a>(&0)
-}

--- a/tests/ui/lifetimes/recover-infer-ret-ty-issue-135845.rs
+++ b/tests/ui/lifetimes/recover-infer-ret-ty-issue-135845.rs
@@ -1,0 +1,11 @@
+// Regression test for #135845.
+
+use std::marker::PhantomData;
+
+fn b<'a>() -> _ {
+    //~^ ERROR the placeholder `_` is not allowed within types on item signatures for return types [E0121]
+    let _: PhantomData<&'a ()> = PhantomData;
+    0
+}
+
+fn main() {}

--- a/tests/ui/lifetimes/recover-infer-ret-ty-issue-135845.stderr
+++ b/tests/ui/lifetimes/recover-infer-ret-ty-issue-135845.stderr
@@ -1,0 +1,15 @@
+error[E0121]: the placeholder `_` is not allowed within types on item signatures for return types
+  --> $DIR/recover-infer-ret-ty-issue-135845.rs:5:15
+   |
+LL | fn b<'a>() -> _ {
+   |               ^ not allowed in type signatures
+   |
+help: replace with the correct return type
+   |
+LL - fn b<'a>() -> _ {
+LL + fn b<'a>() -> i32 {
+   |
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0121`.

